web/src/user_group_edit_members.ts

```diff
@@ -373,8 +509,57 @@ function remove_member({
     do_remove_user_from_group();
 }
 
+function remove_subgroup({
+    group_id,
+    target_subgroup_id,
+    $list_entry,
+}: {
+    group_id: number;
+    target_subgroup_id: number;
+    $list_entry: JQuery;
+}): void {
+    const group = user_groups.get_user_group_from_id(current_group_id);
+    if (!group) {
+        return;
+    }
+
+    function removal_success(): void {
+        if (group_id !== current_group_id) {
+            blueslip.info("Response for subgroup removal came too late.");
+            return;
+        }
+
+        $list_entry.remove();
+        const message = $t({defaultMessage: "Removed successfully."});
+        show_user_group_membership_request_result({
+            message,
+            add_class: "text-success",
+            remove_class: "text-remove",
+        });
+    }
+
+    function removal_failure(): void {
+        show_user_group_membership_request_result({
+            message: $t({defaultMessage: "Error removing subgroup from this group."}),
+            add_class: "text-error",
+            remove_class: "text-success",
+        });
+    }
+
+    function do_remove_subgroup_from_group(): void {
+        edit_user_group_membership({
+            group,
+            removed_subgroups: [target_subgroup_id],
+            success: removal_success,
+            error: removal_failure,
+        });
+    }
+
+    do_remove_subgroup_from_group();
```

Comment:
Why do we define the `do_remove_subgroup_from_group` if it's just going to be called once?